scipy/stats/_fisher_exact.py

```diff
@@ -0,0 +1,505 @@
+            true.
+          * ``conditional_odds_ratio``, float, is the conditional
+            maximum likelihood estimate for the odds ratio.   It is
+            the noncentrality parameter of Fisher's noncentral
+            hypergeometric distribution with the same hypergeometric
+            parameters as ``table`` whose mean is ``table[0, 0]``.
+            (FIXME: I know this need work!)
+          * ``conditional_odds_ratio_ci``, tuple of float, is the
+            conditional exact confidence interval of the odds ratio.
+
+    See Also
+    --------
+    chi2_contingency : Chi-square test of independence of variables in a
+        contingency table.
+
+    Notes
+    -----
+    For tables with large numbers, the (inexact) chi-square test implemented
+    in the function `chi2_contingency` can also be used.
+
+    For backwards compatibility, the object returned by the function
+    also acts like a sequence of length 2 that holds just the
+    ``sample_odds_ratio`` and the ``pvalue`` fields.  The following
+    code to unpack the two values,
+
+        oddsratio, pvalue = fisher_exact(table)
+
+    still works, but that loses the other parameters attached to the
+    return value.  It is recommended to use the return value as an
+    object with attributes, e.g.:
+
+        result = fisher_exact(table)
+        print(result.pvalue, result.conditional_odds_ratio)
+
+    The limits of the confidence interval are the conditional "exact
+    confidence limits" as defined in section 2 of Cornfield [2]_, and
+    originally described by Fisher [1]_.  The conditional odds ratio
+    and confidence interval are also discussed in Section 4.1.2 of the
+    text by Sahai and Khurshid [3]_.
+
+    References
+    ----------
+    .. [1] R. A. Fisher (1935), The logic of inductive inference, Journal
+           of the Royal Statistical Society, Vol. 98, No. 1, pp. 39-82.
+    .. [2] J. Cornfield (1956), A statistical problem arising from
+           retrospective studies. In Neyman, J. (ed.), Proceedings of
+           the Third Berkeley Symposium on Mathematical Statistics and
+           Probability 4, pp. 135-148.
+    .. [3] H. Sahai and A. Khurshid (1996), Statistics in Epidemiology:
+           Methods, Techniques, and Applications, CRC Press LLC, Boca
+           Raton, Florida.
+
+    Examples
+    --------
+    Say we spend a few days counting whales and sharks in the Atlantic and
+    Indian oceans. In the Atlantic ocean we find 8 whales and 1 shark, in
+    the Indian ocean 2 whales and 5 sharks. Then our contingency table is::
+
+                Atlantic  Indian
+        whales     8        2
+        sharks     1        5
+
+    We use this table to find the p-value:
+
+    >>> from scipy.stats import fisher_exact
+    >>> result = fisher_exact([[8, 2], [1, 5]])
+    >>> result.pvalue
+    0.03496503496503495
+
+    The probability that we would observe this or an even more imbalanced
```

Comment:
"The probability under the null hypothesis that ..." would be helpful.

Reply:
Sounds good, I'll add that.